www/front_src/src/Header/userMenu/index.tsx

```diff
@@ -0,0 +1,340 @@
+    if (refreshTimeout.current) {
+      clearTimeout(refreshTimeout.current);
+    }
+    refreshTimeout.current = setTimeout(() => {
+      loaduserData();
+    }, 60000);
+  };
+
+  const toggle = (): void => {
+    setToggled(!toggled);
+  };
+
+  const onCopy = (): void => {
+    if (autologinNode && autologinNode.current) {
+      autologinNode.current.select();
+      window.document.execCommand('copy');
+      setCopied(true);
+    }
+  };
+
+  const handleClick = (e): void => {
+    if (!profile.current || profile.current.contains(e.target)) {
+      return;
+    }
+    setToggled(false);
+  };
+
+  if (!data) {
+    return <MenuLoader width={21} />;
+  }
+
+  const allowEditProfile = allowedPages?.includes(EDIT_PROFILE_TOPOLOGY_PAGE);
+
+  const gethref = window.location.href;
+  const conditionnedhref = gethref + (window.location.search ? '&' : '?');
+  const autolink = `${conditionnedhref}autologin=1&useralias=${data.username}&token=${data.autologinkey}`;
+
+  return (
+    <div className={classnames(classes.wrapRightUser)}>
+      <div className={classnames(classes.wrapRightUserItems)}>
+        <Clock />
+        <div ref={profile as React.RefObject<HTMLDivElement>}>
+          <UserIcon
+            className={classnames(classes.userIcon)}
+            fontSize="large"
+            onClick={toggle}
+          />
+          <div
+            className={classnames(classes.subMenu, {
+              [classes.subMenuActive]: toggled,
+            })}
+          >
+            <div>
+              <ul className={classnames(classes.listUnstyled)}>
+                <li>
+                  <div
+                    className={classnames(
+                      classes.itemLink,
+                      classes.nameAliasContainer,
+                    )}
+                  >
+                    <div>
+                      <Typography className={classes.fullname} variant="body2">
+                        {data.fullname}
+                      </Typography>
+                      <Typography
+                        style={{ wordWrap: 'break-word' }}
+                        variant="body2"
+                      >
+                        {t('as')}
+                        {` ${data.username}`}
+                      </Typography>
+                    </div>
+                    {allowEditProfile && (
+                      <Link
+                        submenuUserEdit
+                        className={classnames(classes.submenuUserEdit)}
+                        to={`/main.php?p=${EDIT_PROFILE_TOPOLOGY_PAGE}&o=c`}
+                        onClick={toggle}
+                      >
+                        <Typography variant="body2">
+                          {t('Edit profile')}
+                        </Typography>
+                      </Link>
+                    )}
+                  </div>
+                </li>
+                {data.autologinkey && (
+                  <div className={classnames(classes.subMenuItemContent)}>
```

Comment:
It might make sense to use the same style as the buttons in the poller menu to stay consistent